identity/src/main/java/com/stripe/android/identity/camera/IdentityScanFlow.kt

```diff
@@ -94,26 +97,34 @@ internal class IdentityScanFlow(
 
             requireNotNull(aggregator).bindToLifecycle(lifecycleOwner)
 
-            analyzerPool =
-                AnalyzerPool.of(
-                    if (parameters == IdentityScanState.ScanType.SELFIE) {
-                        FaceDetectorAnalyzer.Factory(
-                            requireNotNull(faceDetectorModelFile) {
-                                "Failed to initialize FaceDetectorAnalyzer, " +
-                                    "faceDetectorModelFile is null"
-                            },
-                            modelPerformanceTracker
-                        )
-                    } else {
-                        IDDetectorAnalyzer.Factory(
-                            idDetectorModelFile,
-                            verificationPage.documentCapture.models.idDetectorMinScore,
-                            modelPerformanceTracker,
-                            laplacianBlurDetector,
-                            identityAnalyticsRequestFactory,
-                        )
-                    }
-                )
+            try {
+                analyzerPool =
+                    AnalyzerPool.of(
+                        if (parameters == IdentityScanState.ScanType.SELFIE) {
+                            FaceDetectorAnalyzer.Factory(
+                                requireNotNull(faceDetectorModelFile) {
+                                    "Failed to initialize FaceDetectorAnalyzer, " +
+                                        "faceDetectorModelFile is null"
+                                },
+                                modelPerformanceTracker
+                            )
+                        } else {
+                            IDDetectorAnalyzer.Factory(
+                                idDetectorModelFile,
+                                verificationPage.documentCapture.models.idDetectorMinScore,
+                                modelPerformanceTracker,
+                                laplacianBlurDetector,
+                                identityAnalyticsRequestFactory,
+                            )
+                        }
+                    )
+            } catch (e: IllegalStateException) {
+                withContext(Dispatchers.Main) {
+                    errorHandler?.invoke(e) ?: throw e
```

Comment:
This is always going to crash the app if we throw. It feels like we should move every usage over to require an error handler.

Reply:
Got it. This is intended to replicate existing behavior but I agree. I will look into it

Reply:
Updated, could you sanity check the `errorHandler` implementations?